bundler/lib/bundler/lockfile_parser.rb

```diff
@@ -235,19 +235,14 @@ def parse_checksum(line)
 
       version = Gem::Version.new(version)
       platform = platform ? Gem::Platform.new(platform) : Gem::Platform::RUBY
-      source = "#{@lockfile_path}:#{@pos} in the CHECKSUMS lockfile section"
-      checksums = checksums.split(",").map do |c|
-        algo, digest = c.split("-", 2)
-        Checksum::Single.new(algo, digest, source)
-      end
-
+      checksums = Checksum.from_lock(checksums, "#{@lockfile_path}:#{@pos} in the CHECKSUMS lockfile section")
       full_name = GemHelpers.spec_full_name(name, version, platform)
 
       # Don't raise exception if there's a checksum for a gem that's not in the lockfile,
       # we prefer to heal invalid lockfiles
       return unless spec = @specs[full_name]
```

Comment:
This should change order so we don't parse checksums for a not-found spec. Just save the time and return early.